### PR TITLE
Use the new muti-region generating method in stats

### PR DESCRIPTION
**Pull request by valeriuo**
Fix a memory deallocation conflict introduced by https://github.com/samtools/htslib/pull/836

### CODE REVIEW

File: stats.c [hunk 1]
```diff
@@ -2426,80 +2433,45 @@ int main_stats(int argc, char *argv[])
...
-                        if ( all_stats->nregions && all_stats->regions ) {
-                            while (sam_itr_multi_next(info->sam, iter, bam_line) >= 0) {
-                               if (info->split_tag) {
-                                   curr_stats = get_curr_split_stats(bam_line, split_hash, info, targets);
-                                   collect_stats(bam_line, curr_stats, read_pairs);
-                               }
-                               collect_stats(bam_line, all_stats, read_pairs);
-                            }
+                if ( all_stats->nregions && all_stats->regions ) {
+                    while (sam_itr_next(info->sam, iter, bam_line) >= 0) {
```
Comment (daviesrob) [inline stats.c]: Need to check for return value < -1 here

File: stats.c [hunk 2]
```diff
@@ -2426,80 +2433,45 @@ int main_stats(int argc, char *argv[])
...
                     }
-                } else {
-                    fprintf(stderr, "Creation of the region list failed.");
                 }
-
-                hts_idx_destroy(bam_idx);
+                hts_itr_multi_destroy(iter);
             } else {
-                fprintf(stderr, "Random alignment retrieval only works for indexed BAM files.\n");
+                fprintf(stderr, "Multi-region iterator could not be created\n");
```
Comment (daviesrob) [inline stats.c]: Probably want to return an error here

File: stats.c [hunk 3]
```diff
@@ -2426,80 +2433,45 @@ int main_stats(int argc, char *argv[])
...
             }
-
-            bed_destroy(region_hash);
+            hts_idx_destroy(bam_idx);
         } else {
-            fprintf(stderr, "Creation of the region hash table failed.\n");
+            if (has_index_file)
+                fprintf(stderr, "Invalid index file '%s'\n", bam_idx_fname);
+            fprintf(stderr, "Random alignment retrieval only works for indexed files\n");
```
Comment (daviesrob) [inline stats.c]: Missing else?
Also need to return an error.

Comment (valeriuo) [reply stats.c]: > Missing else?

No, just that, when the user specifies the index file, a hint could be given about the validity of the file.

**Pull request merged**